robottests/test_remotescript.robot

```diff
@@ -2,31 +2,241 @@
+
+RemoteScript Create Directory In Target
+    Better Create Directory In Target
+    ${ret2}=    Execute Command In Target   stat /tmp/target_dir
+    Should Be Equal As Integers     ${ret2.status}  0   ${ret2}
+    [Teardown]  Execute Command In Target rm -r /tmp/target_dir
+
+RemoteScript Remove Directory In Target
+    Better Create Directory In Target
+    ${ret2}=    Remove Directory In Target  path=/tmp/target_dir
+    Should Be Equal As Integers     ${ret2.status}  0   ${ret2}
+    ${ret3}=    Execute Command In Target   stat /tmp/target_dir
+    Should Be Equal As Integers     ${ret3.status}  1   ${ret3}
+
+RemoteScript Set Target With Sshkeyfile
+    Set Target With Sshkeyfile  host=${HOST2.host}
+    ...                         username=${HOST2.user}
+    ...                         sshkeyfile=${HOST2.key}
+    ...                         name=host2
+    ${ret}=     Execute Command In Target   echo out    target=host2
+    Should Be Equal As Integers     ${ret.status}   0   ${ret}
+
+RemoteScript Get Target Properties
+    ${ret}=    Get Target Properties  target=default
+    Dictionary Should Contain Sub Dictionary     ${ret}   ${DEFAULT_TARGET}
+
+RemoteScript Execute Script In Target
+    Better Create Directory In Target
+    ${out}=     Run Process  ls ${CURDIR}  shell=${True}
+    Log  ${out.stdout}
+    ${ret2}=    Copy File To Target  source_file=${CURDIR}/target_script.sh  destination_dir=/tmp/target_dir
+    Should Be Equal As Integers     ${ret2.status}  0
+    ${ret3}=     Execute Script In Target    file=/tmp/target_dir/target_script.sh
+    Should Be Equal As Integers     ${ret3.status}  0
+    [Teardown]  Execute Command In Target  rm -r /tmp/target_dir
+
+RemoteScript Execute Background Command In Target
+	FOR		${exec_id}	IN	bg1  bg2  bg3
+			Execute Background Command In Target	command=echo Hello;echo World!>&2
+			...										target=default exec_id=${exec_id}
+	END
+	FOR		${exec_id}	IN	bg1  bg2  bg3
+		${result}=	Wait Background Execution	${exec_id}
+		Should Be Equal As Integers		${result.status}	0
+		Should Be Equal		${result.stdout}	Hello
+		Should Be Equal		${result.stderr}	World!
+	END
+
+RemoteScript Execute Background Script In Target
+    Better Create Directory In Target
+    ${out}=     Run Process  ls ${CURDIR}  shell=${True}
+    Log  ${out.stdout}
+    ${ret2}=    Copy File To Target  source_file=${CURDIR}/target_script.sh  destination_dir=/tmp/target_dir
+    Should Be Equal As Integers     ${ret2.status}  0
+    Execute Background Script In Target  file=/tmp/target_dir/target_script.sh  target=default  exec_id=bg1
+    ${result_bg1}=  Wait Background Execution   exec_id=bg1
+    Log  ${result_bg1}
+    Should Be Equal As Integers     ${result_bg1.status}   0
+    [Teardown]  Execute Command In Target   rm -r /tmp/target_dir
+
+RemoteScript Copy File Between Targets
+    RemoteScript Set Target With Sshkeyfile
+    Better Create Directory In Target
+    Better Create Directory In Target  target=host2
```

Comment:
Create [Setup] for keyword which calls new keyword e.g. 'Setup File Copying Between Targets'.